PDF report fails due to invalid value type

Sentry 305026

```
AssertionError: Expected value to be a str
  File "openforms/submissions/tasks/pdf.py", line 38, in generate_submission_report
    submission_report.generate_submission_report_pdf()
  File "openforms/submissions/models.py", line 761, in generate_submission_report_pdf
    html_report, pdf_report = render_to_pdf(
...
  File "openforms/formio/formatters/default.py", line 227, in format
    return get_value_label(component["values"], value)
  File "openforms/formio/formatters/default.py", line 24, in get_value_label
    assert isinstance(value, str), "Expected value to be a str"
```

Happens when you have a radio-component with an option that has a numeric value.

![image](https://user-images.githubusercontent.com/96970/160230731-cb1817ca-aa51-48aa-8d61-a1b42f0dd22f.png)

--- a/docs/configuration/registration/zgw.rst
+++ b/docs/configuration/registration/zgw.rst
@@ -7,7 +7,7 @@ ZGW API's
 The `ZGW (Zaakgericht Werken) API's`_ are a suite of REST based Zaak services.
 Open Forms can be configured to access these API's to register form submissions.
 
-.. _`ZGW (Zaakgericht Werken) API's`: https://www.vngrealisatie.nl/producten/api-standaarden-zaakgericht-werken
+.. _`ZGW (Zaakgericht Werken) API's`: https://vng.nl/projecten/zaakgericht-werken-api
 
 .. note::
 

--- a/src/openforms/formio/formatters/default.py
+++ b/src/openforms/formio/formatters/default.py
@@ -1,3 +1,4 @@
+import logging
 from datetime import datetime
 from typing import Any, Dict, Iterable, List, Union
 
@@ -15,13 +16,29 @@ from openforms.plugins.plugin import AbstractBasePlugin
 from ..typing import Component, OptionDict
 from .registry import register
 
+logger = logging.getLogger(__name__)
 
-def get_value_label(possible_values: List[OptionDict], value: str) -> str:
+
+def get_value_label(possible_values: List[OptionDict], value: Union[int, str]) -> str:
+    # From #1466 it's clear that Formio does not force the values to be strings, e.g.
+    # if you use numeric values for the options. They are stored as string in the form
+    # configuration, but the submitted value is a number.
+    # See https://github.com/formio/formio.js/blob/4.12.x/src/components/radio/Radio.js#L208
+    # (unmodified in 4.13) but also normalization in the select component
+    # https://github.com/formio/formio.js/blob/4.12.x/src/components/select/Select.js#L1227
+    _original = value
+    # cast to string to compare against the values
+    if not isinstance(value, str):
+        value = str(value)
+        logger.info(
+            "Casted original value %r to string value %s for option comparison",
+            _original,
+            value,
+        )
+
     for possible_value in possible_values:
         if possible_value["value"] == value:
             return possible_value["label"]
-
-    assert isinstance(value, str), "Expected value to be a str"
 
     return value
 
@@ -223,7 +240,7 @@ class CurrencyFormatter(FormioFormatter):
 
 @register("radio")
 class RadioFormatter(FormioFormatter):
-    def format(self, component: Component, value: str) -> str:
+    def format(self, component: Component, value: Union[str, int]) -> str:
         return get_value_label(component["values"], value)
 
 

--- a/src/openforms/formio/formatters/tests/test_issues.py
+++ b/src/openforms/formio/formatters/tests/test_issues.py
@@ -376,3 +376,91 @@ class IssuesTestCase(TestCase):
         res = format_value(select1, value)
 
         self.assertEqual(res, "")
+
+    def test_issue_1466_assertion_error_string(self):
+        """
+        Regression test for radio inputs with numeric values.
+
+        Github issue #1466, when a crash occurred in the PDF generation because of
+        unexpected non-string option value. This appears to happen when values are
+        numeric and Formio keeps the number datatype instead of converting to string.
+        """
+        component = {
+            "allowCalculateOverride": False,
+            "allowMultipleMasks": False,
+            "attributes": {},
+            "autofocus": False,
+            "calculateServer": False,
+            "calculateValue": "",
+            "clearOnHide": True,
+            "conditional": {"eq": "", "show": None, "when": None},
+            "customClass": "",
+            "customDefaultValue": "",
+            "dataGridLabel": False,
+            "dbIndex": False,
+            "defaultValue": "",
+            "description": "",
+            "disabled": False,
+            "encrypted": False,
+            "errorLabel": "",
+            "fieldSet": False,
+            "hidden": False,
+            "hideLabel": False,
+            "id": "e2n18ig",
+            "input": True,
+            "inputType": "radio",
+            "isSensitiveData": False,
+            "key": "projectmanagement",
+            "label": "Hoe tevreden ben je met ons projectmanagement / consultancy en bijbehorende communicatie?",
+            "labelPosition": "top",
+            "modalEdit": False,
+            "multiple": False,
+            "overlay": {"height": "", "left": "", "style": "", "top": "", "width": ""},
+            "persistent": True,
+            "placeholder": "",
+            "prefix": "",
+            "properties": {},
+            "protected": False,
+            "redrawOn": "",
+            "refreshOn": "",
+            "registration": {"attribute": ""},
+            "showCharCount": False,
+            "showInEmail": False,
+            "showWordCount": False,
+            "suffix": "",
+            "tabindex": "",
+            "tableView": False,
+            "tooltip": "",
+            "type": "radio",
+            "unique": False,
+            "validate": {
+                "custom": "",
+                "customPrivate": False,
+                "multiple": False,
+                "onlyAvailableItems": False,
+                "plugins": [],
+                "required": True,
+                "strictDateValidation": False,
+                "unique": False,
+            },
+            "validateOn": "change",
+            "values": [
+                {"label": "1 - Totaal niet tevreden", "value": "1"},
+                {"label": "2", "value": "2"},
+                {"label": "3", "value": "3"},
+                {"label": "4", "value": "4"},
+                {"label": "5", "value": "5"},
+                {"label": "6", "value": "6"},
+                {"label": "7", "value": "7"},
+                {"label": "8", "value": "8"},
+                {"label": "9", "value": "9"},
+                {"label": "10 - Heel erg tevreden", "value": "10"},
+                {"label": "n.v.t.", "value": "nvt"},
+            ],
+            "widget": None,
+        }
+        value = 10
+
+        res = format_value(component, value)
+
+        self.assertEqual(res, "10 - Heel erg tevreden")

